### DISCUSSION TITLE
How to display stop text

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#455] [Help]
Originally posted 2014-11-28 04:04:28.446000 by mickberg in GrandOrgue discussion forum on SourceForge<br><br>Hi.
Sorry to ask such a basic question, but I&#039;m stumped. I am customizing an ODF that had graphics for the stops. I want to use the default GO interface. I have removed the image references, and everything should be OK but I don&#039;t see the label on the stop. What very basic thing am I missing?
Thanks,
Mick Berg.

**Comment by larspalo** [comment]
Originally posted 2014-11-28 08:04:48.964000 by [Martin Koegler](https://github.com/e9925248) in GrandOrgue discussion forum on SourceForge<br><br>Delete TextBreakWidth and/or DispLabelText

**Reply by larspalo** [reply]
Originally posted 2014-11-29 07:36:06.642000 by mickberg in GrandOrgue discussion forum on SourceForge<br><br>There you go! 
Thanks,
Mick.

**Reply by larspalo** [reply]
Originally posted 2014-11-29 07:36:09.935000 by mickberg in GrandOrgue discussion forum on SourceForge<br><br>There you go! 
Thanks,
Mick.

**Comment by Bumblebee001** [comment]
Hi,

I noticed that the stop names appear as text superimposed over the labels I created for each stop. Up to GO 3.1.2330 the console appears as it should. In all subsequent versions the same odf brings up the names of stops over the graphics. No matter what I did to the odf (Display=N; removal of reference to such text; adding a ; at the beginning of the stop name in the odf) the text still appears even after deleting cache and reloading the organ.

What has changed since 2330 that annoyingly shows the text mentioned?

![image](https://user-images.githubusercontent.com/6999990/146950160-4ac5a127-7d91-4f90-b516-88a8a469fa55.png)

**Reply by larspalo** [reply]
Do you have TextBreakWidth=0 set for the displayed stops?

**Comment by larspalo** [comment]
Generally speaking, there are a few different properties that can/should be set for displaying stop names in different ways, depending on what you want to achieve. 

1. For any button/stop the Name= property should always be set with a descriptive name since it will be used in lists (like in the Organ dialog). If nothing else is set, this is also displayed as name if the stop/button is displayed.
2. The DispLabelText= can be used to display a different version of the name (for instance shortened to fit the image). If set, it will override the Name= value for panel display.
3. The TextBreakWidth= is normally used to decide how long a line of text can be before a line break occur. However, if set to 0, it can be used to "hide" any name text.

**Reply by Bumblebee001** [reply]
I did not have this; but after adding it and setting to 0, the console has not changed.... i.e. the text still shows

![image](https://user-images.githubusercontent.com/6999990/146982313-e447d52d-e1bd-4da0-8141-2b1a50517993.png)

**Reply by larspalo** [reply]
I think I'd need to see the whole odf to be able to spot the issue, is it available somewhere? It's not possible that you have separate labels for the stop names?

I'm fairly sure the TextBreakWidth feature in GO works as it should, otherwise both Bureå funeral chapel and Piteå MHS would look wrong too as they rely on it to just display the images without any text overlay.

**Reply by Bumblebee001** [reply]
It was perfectly fine with the old GO installation (3.1.2330). The problem arose with the newer GO installations and only discovered the issue today by pure chance when I installed the latest and loaded the latest sampleset of mine. No one reported any issues and I can only assume that people are still using your version/s rather than Oleg's ones. I have not touched the sample-set since I published it last year.

I shall send you a link to your private email address.

**Reply by larspalo** [reply]
Ok, thanks. I'll look into it when I get it.

**Reply by larspalo** [reply]
Well, the sampleset looks just perfect here! I attach a screenshot below. But it seems that you have some extended version that might be different...
![Skärmbild från 2021-12-21 21-18-23](https://user-images.githubusercontent.com/6173941/146993146-34de6699-9c86-4777-9754-75940c1a5e28.png)

**Reply by Bumblebee001** [reply]
Also... I noticed that I had a piston font size error set at 8 instead of 10 (for I/P). I corrected it. The latest GO will not display it correctly whereas 3.1.2330 does. Something seriously wrong with the latest GO versions handling text in general.

**Reply by Bumblebee001** [reply]
You are right...... I think I have been look at the wrong set. Just noticed that!!!!

**Reply by Bumblebee001** [reply]
Having said that, I was opening the same set on different versions of GO, though and they are showing up differently. What I was doing wrong was that I was changing things in the odf of the original rather than that of the extended. [Effects of old age creeping in :-( ]

**Reply by larspalo** [reply]
No, actually... You have the names still displayed. It's just that the font size is set to 1, and it's black... When I increase it to 14 it's clearly visible. But adding TextBreakWidth=0 hides it alright here!

**Reply by Bumblebee001** [reply]
Same thing again with the original .... stop names appearing in 3.4.4-1 that do not show up in 3.1.2330

![image](https://user-images.githubusercontent.com/6999990/146994874-eaef60a8-368b-4651-bf4d-d6426fd9c4dd.png)

**Reply by Bumblebee001** [reply]
How can one account for the difference between what I am seeing on my computer versus what you are getting when loading it on yours? Are you using the latest GO version (3.4.4-1)?

**Reply by Bumblebee001** [reply]
I notice that the text is in duplicate. TextBreakWidth=0 removed one of those. I must have another entry that's producing the second one.

So my conclusion is that the new versions of GO are more sensitive/ specific to what's in the odf. If the instruction is absent (in this case the TextBreakWidth=0) it sets a default value of some sort (in this case shows up the text).

**Reply by larspalo** [reply]
So, at least you have a solution in TextBreakWidth=0 to hide the text. That's better than setting font size to 1.

**Reply by Bumblebee001** [reply]
So bottom line: I modified my odfs accordingly so that the consoles in both the original and the extended versions appear on the latest GO versions the way they did in the older v3.1.2330. (I must check my other VPOs...... )
